code/game/machinery/Sleeper.dm

```diff
@@ -459,19 +465,16 @@
 	else
 		to_chat(user, "There's no occupant in the sleeper!")
 
-/obj/machinery/sleeper/verb/eject()
-	set name = "Eject Sleeper"
-	set category = "Object"
-	set src in oview(1)
-
-	if(usr.default_can_use_topic(src) != STATUS_INTERACTIVE)
-		return
-	if(usr.incapacitated()) //are you cuffed, dying, lying, stunned or other
+/obj/machinery/sleeper/AltClick(mob/user)
+	if(issilicon(user))
+		eject()
```

Comment:
could you add a return after this? Its going to return anyway in the next check so we might aswell avoid checking adjacency